src/backend/opencl/platform.cpp

```diff
@@ -233,13 +235,13 @@ int getDeviceIdFromNativeId(cl_device_id id) {
 
     common::lock_guard_t lock(devMngr.deviceMutex);
 
-    int nDevices = devMngr.mDevices.size();
-    int devId    = 0;
+    size_t nDevices = devMngr.mDevices.size();
+    size_t devId    = 0;
     for (devId = 0; devId < nDevices; ++devId) {
         if (id == devMngr.mDevices[devId]->operator()()) { break; }
     }
 
-    return devId;
+    return static_cast<int>(devId);
```

Comment:
nit pick: Perhaps just do `static_cast` on return of `mDevices.size()` itself, that way no need alter other lines.

Reply:
Just incredible.  This MSVC reports, and nothing for the previous bigger ones.

I changed everything to int with a static_cast as suggested.

Reply:
Perhaps you missed that line in the huge lot of warnings, to be fair - we do have quite a few warnings that needs to be addressed with MSVC compiler. We largely develop using GCC/clang, so we often miss the MSVC warnings unless we test/debug on windows for specific issues.